### PR TITLE
HIVE-28081: Code refine on ClearDanglingScratchDir::removeLocalTmpFiles

### PR DESCRIPTION
**Pull request by zhangbutao**
### Why are the changes needed?

Code refine. Make code more readable.

### Does this PR introduce _any_ user-facing change?

No
### Is the change a dependency upgrade?

No

### How was this patch tested?

Existing UT.

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/session/ClearDanglingScratchDir.java [hunk 1]
```diff
@@ -252,17 +252,12 @@ static Options createOptions() {
...
       for (File file : files) {
-        success = false;
-        if (file.canWrite()) {
-          success = file.delete();
-        }
-        if (success) {
+        if (file.canWrite() && file.delete()) {
           consoleMessage("While removing '" + sessionName + "' dangling scratch dir from HDFS, "
                   + "local tmp session file '" + file.getPath() + "' has been cleaned as well.");
-        } else if (file.getName().startsWith(sessionName)) {
```
Comment (zhangbutao) [inline ql/src/java/org/apache/hadoop/hive/ql/session/ClearDanglingScratchDir.java:265]: See Line 254, file in here is already` startsWith(sessionName)`, so no need check this again.

File: ql/src/java/org/apache/hadoop/hive/ql/session/ClearDanglingScratchDir.java [hunk 2]
```diff
@@ -252,17 +252,12 @@ static Options createOptions() {
    */
   private void removeLocalTmpFiles(String sessionName, String localTmpdir) {
     File[] files = new File(localTmpdir).listFiles(fn -> fn.getName().startsWith(sessionName));
-    boolean success;
     if (files != null) {
```
[the remark this answers is not in the document]
Comment (zhangbutao) [reply ql/src/java/org/apache/hadoop/hive/ql/session/ClearDanglingScratchDir.java:255]: There is no code after the `if` clause, so we don't need to add extra return. :)

[the remark this answers is not in the document]
Comment (okumin) [reply ql/src/java/org/apache/hadoop/hive/ql/session/ClearDanglingScratchDir.java:255]: Sorry for the lack of words. I expected the following return would let us know we do nothing on `else` a bit earlier.

```
if (files == null) {
  return;
}
```

**Review by zabetak — APPROVED** [review]
LGTM!

**Pull request merged**